.evergreen/buildvariants.in.yml

```diff
@@ -11,30 +18,145 @@ buildvariants:
+    run_on: 'windows-vsCurrent-large',
+  },
+  {
+    name: 'test-and-package-rhel',
+    display_name: 'RHEL 8.0',
+    run_on: 'rhel80-large',
+  },
+  {
+    name: 'test-and-package-macos-x64',
+    display_name: 'MacOS x64 11',
+    run_on: 'macos-1100',
+    run_on_gui: 'macos-1100-gui'
+  },
+  {
+    name: 'test-and-package-macos-arm',
+    display_name: 'MacOS arm64 11',
+    run_on: 'macos-1100-arm64',
+    run_on_gui: 'macos-1100-arm64-gui'
+  }
+];
+
+for (const buildVariant of testAndPackageBuildVariants) { %>
+  - name: <%= buildVariant.name %>
+
+    display_name: <%= buildVariant.display_name %> (Test and Package)
+    run_on: <%= buildVariant.run_on %>
+    tasks:
+    - name: check
+    - name: test
+    - name: test-electron
+      <% if (buildVariant.run_on_gui) { %>
+      run_on: <%= buildVariant.run_on_gui %>
+      <% } %>
+    - name: package-compass
+    - name: package-compass-readonly
+    - name: package-compass-isolated
+<% } %>
+
+<%
+const testPackagedAppBuildVariants = [
+  {
+    name: 'test-packaged-app-ubuntu',
+    display_name: 'Ubuntu 20.04',
+    run_on: 'ubuntu1604-large',
+    depends_on: 'test-and-package-ubuntu',
+    test_latest_alpha: true
+  },
+  {
+    name: 'test-packaged-app-windows',
+    display_name: 'Windows 10',
+    run_on: 'windows-vsCurrent-large',
+    depends_on: 'test-and-package-windows',
+    test_latest_alpha: true
+  },
+  {
+    name: 'test-packaged-app-rhel',
+    display_name: 'RHEL 8.0',
+    run_on: 'rhel80-large',
+    depends_on: 'test-and-package-rhel'
+  },
+  {
+    name: 'test-packaged-app-macos-11-x64',
+    display_name: 'MacOS x64 11',
+    run_on: 'macos-1100-gui',
+    patchable: false,
+    depends_on: 'test-and-package-macos-x64'
+  },
+  {
+    name: 'test-packaged-app-macos-11-arm',
+    display_name: 'MacOS arm64 11',
+    run_on: 'macos-1100-arm64-gui',
+    patchable: false,
+    depends_on: 'test-and-package-macos-arm'
+  },
+  {
+    name: 'test-packaged-app-macos-14-x64',
+    display_name: 'MacOS x64 14',
+    run_on: 'macos-14-gui',
+    patchable: false,
+    depends_on: 'test-and-package-macos-x64'
+  },
+  {
+    name: 'test-packaged-app-macos-14-arm',
+    display_name: 'MacOS arm64 14',
+    run_on: 'macos-14-arm64-gui',
+    depends_on: 'test-and-package-macos-arm'
+  },
+];
+
+for (const buildVariant of testPackagedAppBuildVariants) { %>
+  - name: <%= buildVariant.name %>
+    display_name: <%= buildVariant.display_name %> (Test Packaged App)
+    run_on: <%= buildVariant.run_on %>
+    patchable: <%= buildVariant.patchable ?? true %>
+    depends_on:
+      - name: package-compass
+        variant: <%= buildVariant.depends_on %>
+    tasks:
+      - name: test-packaged-app-60x-community
+      - name: test-packaged-app-60x-enterprise
+      - name: test-packaged-app-70x-community
+      - name: test-packaged-app-70x-enterprise
+      - name: test-packaged-app-80x-community
+      - name: test-packaged-app-80x-enterprise
```

Comment:
I think we might want to limit this to just latest stable on patches for macos too (this was the case for a long time, and seems like only recently we added more variants here), we really don't have enough machines to have such a big matrix for macos and not create a bottleneck

Reply:
Would it be an issue to test only oldest and latest everywhere for patches?

Reply:
This sounds like a very good approach, yeah!